doc/user/monitoring.md

```diff
@@ -0,0 +1,39 @@
+# Monitoring
+
+## Expose metrics port
+
+The following example exposes `/metrics` port by creating a `Service` object.
```

Comment:
We probably need a bit of context on what the `metrics` port is.
For e.g the fact that the operator will listen and serve metrics on the TCP address on `:60000/metrics` by default:
```Go
http.Handle("/"+k8sutil.PrometheusMetricsPortName, promhttp.Handler())
go http.ListenAndServe(":"+strconv.Itoa(k8sutil.PrometheusMetricsPort), nil)
```
https://github.com/operator-framework/operator-sdk/blob/master/pkg/sdk/metrics.go#L29-L30
https://github.com/operator-framework/operator-sdk/blob/master/pkg/util/k8sutil/constants.go#L30-L34

And the fact that by default the operator container exposes that port with the portname `metrics`:
```YAML
apiVersion: apps/v1
kind: Deployment
metadata:
  name: memcached-operator
spec:
  replicas: 1
  selector:
    matchLabels:
      name: memcached-operator
  template:
...
    spec:
      containers:
        - name: memcached-operator
          ports:
          - containerPort: 60000
            name: metrics
```

But if https://github.com/kubernetes-sigs/controller-runtime/pull/132 gets merged soon(hopefully since it looks complete) then this doc will change significantly in how we set the tcp address or port number that the metrics are exposed on.
So I'm thinking that we might hold off writing this doc until we switch over to using the controller-runtime and that PR lands.